src/tree/mod.rs

```diff
@@ -700,14 +700,14 @@ impl<T> Tree<T> {
     {
         let lower_upper_test = self.find_subtree_root_between_ids(first_id, second_id)
             .map(|_| (first_id, second_id))
-            .or(self.find_subtree_root_between_ids(second_id, first_id)
+            .or_else(|| self.find_subtree_root_between_ids(second_id, first_id)
```

Comment:
@Drakulix since you originally wrote this bit I wanted to call attention to it.  First, I just wanted to make sure that you don't see any issues that could arise from doing this, but I also wanted to let you know about the difference between these two if you're not already aware (I just learned about it today).

As it turns out, the value in `.or()` will always be evaluated, even if the option that it is being called on is a `Some()` value.  The `or_else()` method however, will only call the provided closure if the previous option is a `None`.  Doing the above, we can avoid an extra call to `find_subtree_root_between_ids()` in the case where the first call returned a `Some()` value.